cmd/main.go

```diff
@@ -2,246 +2,109 @@ package main
 	}
 
-	Run(ctx, args, nil)
-}
-
-func Run(ctx context.Context, args RuntimeArgs, _ chan<- int) {
-	log.Logger = log.Output(zerolog.ConsoleWriter{Out: os.Stderr}).Level(args.LogLevel)
+	// Setup logging
+	ctx := gosdk.SetupLogger(context.Background(), cfg.LogLevel)
 
-	ctx = log.With().Logger().WithContext(ctx)
-
-	// Cancel on SIGINT/SIGTERM too (centralized; no per-runner signal goroutines needed)
+	// signal.NotifyContext provides cancellation on SIGINT/SIGTERM
 	ctx, stop := signal.NotifyContext(ctx, syscall.SIGINT, syscall.SIGTERM)
 	defer stop()
 
-	config := gosdk.MakeAppchainConfig(ChainID, args.MutlichainConfig)
-
-	config.EmitterPort = args.EmitterPort
-	config.AppchainDBPath = args.AppchainDBPath
-	config.EventStreamDir = args.EventStreamDir
-	config.TxStreamDir = args.TxStreamDir
-	config.Logger = &log.Logger
-
-	chainDBs, err := gosdk.NewMultichainStateAccessDB(args.MutlichainConfig)
-	if err != nil {
-		log.Fatal().Err(err).Msg("Failed to create multichain db")
-	}
-
-	msa := gosdk.NewMultichainStateAccess(chainDBs)
-
-	// инициализируем базу на нашей стороне
-	appchainDB, err := mdbx.NewMDBX(mdbxlog.New()).
-		Path(config.AppchainDBPath).
-		WithTableCfg(func(_ kv.TableCfg) kv.TableCfg {
-			return gosdk.MergeTables(
-				gosdk.DefaultTables(),
-				application.Tables(),
-			)
-		}).Open()
-	if err != nil {
-		log.Fatal().Err(err).Msg("Failed to appchain mdbx database")
-	}
-
-	defer appchainDB.Close()
-
-	subs, err := gosdk.NewSubscriber(ctx, appchainDB)
-	if err != nil {
-		log.Fatal().Err(err).Msg("Failed to create subscriber")
-	}
-
-	stateTransition := gosdk.NewBatchProcesser[application.Transaction[application.Receipt]](
-		application.NewStateTransition(msa),
-		msa,
-		subs,
-	)
-
-	localDB, err := mdbx.NewMDBX(mdbxlog.New()).
-		Path(args.LocalDBPath).
-		WithTableCfg(func(_ kv.TableCfg) kv.TableCfg {
-			return txpool.Tables()
-		}).
-		Open()
-	if err != nil {
-		log.Fatal().Err(err).Msg("Failed to local mdbx database")
-	}
-
-	defer localDB.Close()
-
-	// fixme dynamic val set. Right now it is especially for local development with pelacli
-	valset := &gosdk.ValidatorSet{Set: map[gosdk.ValidatorID]gosdk.Stake{0: 100}}
-
-	var epochKey [4]byte
-	binary.BigEndian.PutUint32(epochKey[:], 1)
-
-	valsetData, err := cbor.Marshal(valset)
-	if err != nil {
-		log.Fatal().Err(err).Msg("Failed to marshal validator set data")
-	}
-
-	err = appchainDB.Update(ctx, func(tx kv.RwTx) error {
-		return tx.Put(gosdk.ValsetBucket, epochKey[:], valsetData)
-	})
-	if err != nil {
-		log.Fatal().Err(err).Msg("Failed to appchain mdbx database")
+	if err := Run(ctx, cfg); err != nil {
+		log.Fatal().Err(err).Msg("Failed to run appchain")
 	}
+}
 
-	txPool := txpool.NewTxPool[application.Transaction[application.Receipt]](
-		localDB,
-	)
+// Run starts the appchain with the given config. Exported for testing.
+func Run(ctx context.Context, cfg *gosdk.InitConfig) error {
+	// Add custom tables to config
+	cfg.CustomTables = application.Tables()
```

Comment:
In future we need to change it, like adding a method RegisterTables or something

Reply:
Okay